lib/membrane_rtc_engine/endpoints/hls_endpoint.ex

```diff
@@ -419,46 +412,159 @@ if Enum.all?(
+          link_builder
+          |> to({:track_receiver, track.id})
+          |> to({:depayloader, track.id})
+          |> to({:video_parser, track.id})
+          |> to({:decoder, track.id}, Membrane.H264.FFmpeg.Decoder)
+          |> to({:framerate_converter, track.id})
+          |> via_in(Pad.ref(:input, track.id),
+            options: [initial_placement: initial_placement, timestamp_offset: offset]
+          )
+          |> to(:compositor)
+        ]
+      }
+
+      parent_spec =
+        if track.metadata["mainPresenter"] do
+          parent_spec
+        else
+          blank_placement =
+            VideoLayoutMaker.get_track_layout(
+              :blank,
+              basic_tracks_number,
+              state.mixer_config.video.caps
+            )
+
+          {width, height} = blank_placement.display_size
+
+          blank_spec = %ParentSpec{
+            children: %{
+              {:blank, track.id} => %Membrane.BlankVideoGenerator{
+                caps: %Membrane.RawVideo{
+                  width: width,
+                  height: height,
+                  pixel_format: :I420,
+                  framerate: state.mixer_config.video.caps.framerate,
+                  aligned: true
+                },
+                duration: :infinity
+              },
+              {:realtimer, track.id} => Membrane.Realtimer
+            },
+            links: [
+              link({:blank, track.id})
+              |> to({:realtimer, track.id})
+              |> via_in(Pad.ref(:input, {:blank, track.id}),
+                options: [initial_placement: blank_placement, timestamp_offset: offset]
+              )
+              |> to(:compositor)
+            ]
+          }
+
+          merge_parent_specs(parent_spec, blank_spec)
+        end
+
+      state
+      |> generate_audio_mixer(ctx)
+      |> merge_parent_specs(generate_compositor(state, ctx))
+      |> merge_parent_specs(parent_spec)
+    end
+
     if Enum.all?(@compositor_deps, &Code.ensure_loaded?/1) do
-      defp hls_links_and_children(link_builder, %{encoding: :H264} = track, state, ctx) do
-        parent_spec = %ParentSpec{
+      defp generate_compositor(_state, ctx) when is_map_key(ctx.children, :compositor),
+        do: %ParentSpec{children: %{}, links: []}
+
+      defp generate_compositor(state, _ctx) do
+        compositor = %Membrane.VideoCompositor{
+          caps: state.mixer_config.video.caps,
+          real_time: state.mixer_config.video.real_time
+        }
+
+        video_parser_out = %Membrane.H264.FFmpeg.Parser{
+          alignment: :au,
+          attach_nalus?: true
+          # framerate: state.mixer_config.video.output_framerate
+        }
+
+        {frames_per_second, 1} = state.mixer_config.video.caps.framerate
+        seconds_number = Membrane.Time.as_seconds(state.segment_duration.target)
+
+        initial_placement = %VideoPlacement{
+          position: {0, 0},
+          display_size: {20, 20},
+          z_value: 0.0
+        }
```

Comment:
Shouldn't we pass it as an option?

Reply:
We do in input pad options (in line 551) 
```elixir
|> via_in(:input, options: [initial_placement: initial_placement])
```